### ISSUE TITLE
Locking the vehicle and pressing V renders vehicle unaccessible to everyone

### ISSUE THREAD

**Issue by codexGW** [A3Armory/A3A_A3Wasteland_Sock.Altis#8]
_From @AgentRev on March 4, 2016 6:18_

As the title says, if you have the Lock vehicle action selected while inside the vehicle, and you press spacebar to trigger the action, while also pressing V simultaneously to get out, the vehicle remains locked permanently and nobody can access it.

Maybe save the player UID of whoever triggered the lock, and allow him to unlock the vehicle from the outside in case this happens. Of course, the vehicle should remain lockable from the inside only.

_Copied from original issue: A3Armory/A3A_A3Wasteland_Sock.Stratis#1_

**Comment by codexGW** [comment]
When I get sometime later I will test this with a friend and comment more about this. I don't think we have heard anything about this because anyone with a toolkit can do a picklock and it probably resets after a restart.

**Comment by codexGW** [comment]
_From @AgentRev on March 4, 2016 18:1_

Oh there's picklock? I didn't know.

**Comment by codexGW** [comment]
It's something I added awhile back.
https://github.com/A3Armory/A3A_A3Wasteland_Sock.Stratis/blob/master/addons/scripts/lockPick.sqf
https://github.com/A3Armory/A3A_A3Wasteland_Sock.Stratis/blob/master/client/functions/playerActions.sqf#L27

**Comment by codexGW** [comment]
I just tested this and reproduced it. Thanks for the info Agent. The vehicle re-saving and picklock is probably why no one complained or reported it.

https://github.com/A3Armory/A3A_A3Wasteland_Sock.Altis/blob/dev/addons/vsave/vehicleReSave.sqf#L44

**Comment by AmericanPride** [comment]
There still is a problem when you enter a vehicle, you lock it and disconnect or eject from a plane, the vehicle becomes unusable to everyone. you can't save it, move it, lock pick it or anything even to the person who was in it last and locked it. I can get a video demonstrating it

**Comment by codexGW** [comment]
I will try to get this fixed by doing what Agent suggested. Any help with this would be appreciated.

**Comment by codexGW** [comment]
@AgentRev At a quick glance do you think this will solve this issue or cause any issues with get out and emergency eject?
https://github.com/A3Armory/A3A_A3Wasteland_Sock.Altis/commit/8459648dc82d9c45c2e4f8ee15ac6cf41bb57bf0

**Comment by codexGW** [comment]
I tested it and no errors it seem and it works. I don't think its needed in getOutVehicle.sqf.

**Comment by AgentRev** [comment]
Yes, it is needed.

**Comment by AgentRev** [comment]
In fact, you don't need it in fn_emergencyEject, because getOutVehicle will trigger at the same time nonetheless.

**Comment by tylers-dev** [comment]
Any fix for if a player disconnects while in a locked vehicle?

**Comment by AgentRev** [comment]
Would probably require putting a similar snippet in `HandleDisconnect` with `remoteExec`, but with a check to make sure the player doesn't explicitly own the vehicle; if he does, then leave it locked.

**Comment by codexGW** [comment]
I didn't add the owner check because a disconnect in a vehicle is leaving it and I didn't know how. https://github.com/A3Armory/A3A_A3Wasteland_Sock.Altis/commit/719f7947ac690ecba1732c581e2fe12c742f6c43

**Comment by AgentRev** [comment]
Say I own the car, I get inside and I lock it. Time to leave for the night, I disconnect. I would not expect my own car to get unlocked.

**Comment by AgentRev** [comment]
Also, I'm pretty sure I specifically told you to use remoteExec, not A3W_fnc_MP.

**Comment by codexGW** [comment]
A3W_fnc_MP works why should I not use it? I couldn't figure out how to add a check for owner. What would you add here for that? https://github.com/A3Armory/A3A_A3Wasteland_Sock.Altis/blob/dev/server/init.sqf#L38

If I spawn back it's not inside of the vehicle and I was thinking in case of teammates close by it might be best to have it unlocked. You can still get out and lock it which will save. Most do this because you spawn inside the vehicle glitched with sock if its not moved.

**Comment by AgentRev** [comment]
A3W_fnc_MP is outdated, deprecated, and slow, you shouldn't be using it anymore in brand-new code. Also, the vehicle is undergoing a locality change at the moment of disconnect, there is a slight chance that the MP func tries to target a client that no longer exists, which would still leave it locked.

A vehicle with a claimed ownership belongs solely to his owner, not his teammates. Hence, the game should never unlock a claimed vehicle without the owner's input, unless it's being lockpicked of course.

**Comment by codexGW** [comment]
I would like to add owner. Are you not seeing my question how I would do that? I figure this would be better then not having it. As for teammates, I was thinking if they are doing a mission or op and their ride or heavy armor losses internet they can get back to base and/or get cover.

I tried adding something like this here https://github.com/A3Armory/A3A_A3Wasteland_Sock.Altis/blob/dev/server/init.sqf#L38
`if (_veh != _unit && _veh getVariable ['ownerUID',''] != getPlayerUID _unit) then`
After spending more time then I care to admit, trying to get this to work, I gave up.
